sql/modules/Reconciliation.sql

```diff
@@ -1,15 +1,42 @@
 BEGIN;
 
+DROP FUNCTION IF EXISTS reconciliation__submit_set(int, int[]);
 CREATE OR REPLACE FUNCTION reconciliation__submit_set(
-        in_report_id int, in_line_ids int[]) RETURNS bool AS
+        in_report_id int, in_line_ids int[], in_end_date date
+      ) RETURNS bool AS
 $$
+DECLARE t_balance FLOAT;
+DECLARE t_cleared FLOAT;
 BEGIN
+        SELECT reconciliation__get_cleared_balance(cr.chart_id,cr.end_date)
+        INTO t_cleared
+        FROM cr_report cr
+        WHERE cr.id = in_report_id;
+
+        SELECT CAST(t_cleared - SUM(crl.our_balance) - cr.their_total AS FLOAT) INTO t_balance
+        FROM cr_report_line crl
+        JOIN cr_report cr ON cr.id = crl.report_id
+        WHERE cr.id = in_report_id
+        AND crl.cleared IS true
+        AND crl.clear_time IS NOT NULL
+        GROUP BY cr.their_total;
+
+        -- t_balance is always NULL on RAISE statement
+--        IF t_balance IS NULL OR t_balance <> 0 THEN
+--            RAISE EXCEPTION 'Unbalanced report by %', t_balance;
+--        END IF;
+
         UPDATE cr_report set submitted = true where id = in_report_id;
-        PERFORM reconciliation__save_set(in_report_id, in_line_ids);
+        PERFORM reconciliation__save_set(in_report_id, in_line_ids, in_end_date);
 
         RETURN FOUND;
 END;
 $$ LANGUAGE PLPGSQL;
+COMMENT ON FUNCTION reconciliation__submit_set(
+        in_report_id int, in_line_ids int[], in_end_date date) IS
+$$Submits a reconciliation report for approval.
+in_line_ids is used to specify which report lines are cleared, finalizing the report.
+in_end_date is used to specify the clearing date. Defaults to now.$$;
```

Comment:
All cleared entries are declared cleared at the end of the report. I do not have a problem using a better name